{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "### Problem\n\nI've run into an odd situation where neovim exits while I am editing a file Python, returning me back to the shell without saving . It does not generate any visible error message. It does not clean up the swap file. The swap file is stale, and the latest unsaved changes in the file are lost.\r\n\r\n\n\n### Steps to reproduce\n\nI don't have a good reproducer, and the situation may be specific to the plugins I have installed on my machine (my nvim config is based on `nvim-kickstart`). In other words, feel free to close this commit.\r\n\r\nHowever, I *did* manage to trigger this several times and was able to capture a backtrace in LLDB. I am sharing it here in case it is useful.\r\n```\r\n(lldb) bt\r\n* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.4\r\n  * frame #0: 0x000000018aa99c20 libsystem_c.dylib` exit\r\n    frame #1: 0x0000000100150f34 nvim` os_exit  + 152\r\n    frame #2: 0x00000001000dfbf0 nvim` exit_event.llvm.14166419569618057245  + 128\r\n    frame #3: 0x00000001000dec24 nvim` multiqueue_process_events  + 92\r\n    frame #4: 0x000000010025aab4 nvim` ui_client_run  + 136\r\n    frame #5: 0x0000000100150ac0 nvim` main  + 11164\r\n    frame #6: 0x000000018a883f28 dyld` start  + 2236\r\n(lldb)\r\n```\r\nIn all cases, the sudden exit happened when I was switching from normal to insert mode by pressing the `o` key.\n\n### Expected behavior\n\nThe editor should not simply exit.\n\n### Neovim version (nvim -v)\n\nNVIM v0.10.0-dev-2537+g6d680d977-Homebrew\n\n### Vim (not Nvim) behaves the same?\n\nI don't use plain vim\n\n### Operating system/version\n\n13.6.2\n\n### Terminal name/version\n\nKitty 0.31.0\n\n### $TERM environment variable\n\ntmux-256color\n\n### Installation\n\nBrew",
    "closed_at": "2024-04-11T21:44:52Z",
    "comment_data": [
        {
            "author_association": "MEMBER",
            "body": "Please see https://github.com/neovim/neovim/tree/master/src/nvim#tui-debugging-with-gdblldb for how to obtain a correct backtrace.",
            "created_at": "2024-04-11T20:49:17Z",
            "html_url": "https://github.com/neovim/neovim/issues/28282#issuecomment-2050514152",
            "id": 2050514152,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/28282",
            "node_id": "IC_kwDOAPphoM56OFzo",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2050514152/reactions"
            },
            "updated_at": "2024-04-11T20:49:17Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2050514152",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/35768171?v=4",
                "events_url": "https://api.github.com/users/zeertzjq/events{/privacy}",
                "followers_url": "https://api.github.com/users/zeertzjq/followers",
                "following_url": "https://api.github.com/users/zeertzjq/following{/other_user}",
                "gists_url": "https://api.github.com/users/zeertzjq/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/zeertzjq",
                "id": 35768171,
                "login": "zeertzjq",
                "node_id": "MDQ6VXNlcjM1NzY4MTcx",
                "organizations_url": "https://api.github.com/users/zeertzjq/orgs",
                "received_events_url": "https://api.github.com/users/zeertzjq/received_events",
                "repos_url": "https://api.github.com/users/zeertzjq/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/zeertzjq/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/zeertzjq/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/zeertzjq"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Also, if you are using [nvim-ufo](https://github.com/kevinhwang91/nvim-ufo), try updating its version to after this commit: https://github.com/kevinhwang91/nvim-ufo/commit/e29dbf5b2d76ea43f4e585e3deeb6510488eee3f",
            "created_at": "2024-04-11T20:57:20Z",
            "html_url": "https://github.com/neovim/neovim/issues/28282#issuecomment-2050528284",
            "id": 2050528284,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/28282",
            "node_id": "IC_kwDOAPphoM56OJQc",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2050528284/reactions"
            },
            "updated_at": "2024-04-11T20:57:52Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2050528284",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/35768171?v=4",
                "events_url": "https://api.github.com/users/zeertzjq/events{/privacy}",
                "followers_url": "https://api.github.com/users/zeertzjq/followers",
                "following_url": "https://api.github.com/users/zeertzjq/following{/other_user}",
                "gists_url": "https://api.github.com/users/zeertzjq/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/zeertzjq",
                "id": 35768171,
                "login": "zeertzjq",
                "node_id": "MDQ6VXNlcjM1NzY4MTcx",
                "organizations_url": "https://api.github.com/users/zeertzjq/orgs",
                "received_events_url": "https://api.github.com/users/zeertzjq/received_events",
                "repos_url": "https://api.github.com/users/zeertzjq/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/zeertzjq/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/zeertzjq/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/zeertzjq"
            }
        },
        {
            "author_association": "NONE",
            "body": "Thanks for the advice -- I am not using `nvim-ufo`, however. I updated to the latest `HEAD` revision of neovim and haven't been able to reproduce the crash on this version. So I will close this issue.",
            "created_at": "2024-04-11T21:44:52Z",
            "html_url": "https://github.com/neovim/neovim/issues/28282#issuecomment-2050606019",
            "id": 2050606019,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/28282",
            "node_id": "IC_kwDOAPphoM56OcPD",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2050606019/reactions"
            },
            "updated_at": "2024-04-11T21:44:52Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2050606019",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1203629?v=4",
                "events_url": "https://api.github.com/users/wjakob/events{/privacy}",
                "followers_url": "https://api.github.com/users/wjakob/followers",
                "following_url": "https://api.github.com/users/wjakob/following{/other_user}",
                "gists_url": "https://api.github.com/users/wjakob/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/wjakob",
                "id": 1203629,
                "login": "wjakob",
                "node_id": "MDQ6VXNlcjEyMDM2Mjk=",
                "organizations_url": "https://api.github.com/users/wjakob/orgs",
                "received_events_url": "https://api.github.com/users/wjakob/received_events",
                "repos_url": "https://api.github.com/users/wjakob/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/wjakob/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/wjakob/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/wjakob"
            }
        }
    ],
    "comments": 3,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/28282/comments",
    "created_at": "2024-04-11T20:28:15Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/28282/events",
    "html_url": "https://github.com/neovim/neovim/issues/28282",
    "id": 2238542607,
    "labels": [
        {
            "color": "F9D0C4",
            "default": false,
            "description": "issue reporting a crash or segfault",
            "id": 435854234,
            "name": "bug-crash",
            "node_id": "MDU6TGFiZWw0MzU4NTQyMzQ=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/bug-crash"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/28282/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "I_kwDOAPphoM6FbXMP",
    "number": 28282,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/neovim/neovim/issues/28282/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "closed",
    "state_reason": "completed",
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/28282/timeline",
    "title": "neovim simply exits while editing a file",
    "updated_at": "2024-04-11T21:45:06Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/28282",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/1203629?v=4",
        "events_url": "https://api.github.com/users/wjakob/events{/privacy}",
        "followers_url": "https://api.github.com/users/wjakob/followers",
        "following_url": "https://api.github.com/users/wjakob/following{/other_user}",
        "gists_url": "https://api.github.com/users/wjakob/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/wjakob",
        "id": 1203629,
        "login": "wjakob",
        "node_id": "MDQ6VXNlcjEyMDM2Mjk=",
        "organizations_url": "https://api.github.com/users/wjakob/orgs",
        "received_events_url": "https://api.github.com/users/wjakob/received_events",
        "repos_url": "https://api.github.com/users/wjakob/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/wjakob/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/wjakob/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/wjakob"
    }
}